src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIService.java

```diff
@@ -325,6 +327,32 @@ public void updatePlanRepository(String buildProjectKey, String buildPlanKey, St
         log.error("Unsupported action: GitLabCIService.updatePlanRepository()");
     }
 
+    @Override
+    public List<Long> getAllArtemisBuildPlanServerNotificationIds(String buildPlanKey) {
+        return null;
+    }
+
+    @Override
+    public void deleteBuildPlanServerNotificationId(String buildPlanKey, Long serverNotificationId) {
+
+    }
+
+    @Override
+    public void createBuildPlanServerNotification(String buildPlanKey, String serverNotificationUrl) {
+
+    }
+
+    @Override
+    public void fixBuildPlanNotification(String projectKey, String buildPlanKey, VcsRepositoryUrl repositoryUrl) {
+        try {
+            updateVariable(getRepositoryPath(repositoryUrl), VARIABLE_NOTIFICATION_URL_NAME, artemisServerUrl.toExternalForm() + NEW_RESULT_RESOURCE_API_PATH);
+        }
+        catch (GitLabApiException e) {
+            log.error("Error while updating the notification url for the build plan " + buildPlanKey, e);
+            throw new RuntimeException(e);
```

Comment:
`GitlabCIException` extends `ContinuousIntegrationException` extends `RuntimeException`.
Similar exceptions exist for Jenkins and Bamboo, too, if needed there.